source/client.c

```diff
@@ -173,6 +173,40 @@ static void s_init_statistics(struct aws_mqtt_connection_operation_statistics_im
     aws_atomic_store_int(&stats->unacked_operation_size_atomic, 0);
 }
 
+static bool s_is_topic_shared_topic(struct aws_byte_cursor *input) {
+    char *input_str = (char *)input->ptr;
+    if (strncmp("$share", input_str, strlen("$share")) == 0) {
+        return true;
+    }
+    return false;
+}
+
+static struct aws_string *s_get_normal_topic_from_shared_topic(struct aws_string *input) {
+    const char *input_char_str = aws_string_c_str(input);
+    int input_char_length = strlen(input_char_str);
+    int found_slashes = 0;
+    int split_position = 0;
+    while (split_position < input_char_length) {
+        split_position += 1;
+        if (input_char_str[split_position] == '/') {
+            found_slashes += 1;
+            if (found_slashes == 2) {
+                break;
+            }
+        }
+    }
+    // A properly formatted shared subscription topic will be "$share/<group id>/<topic>" and that means there needs
+    // to be 2 slashes at least. The loop above stops at 2, so if it is NOT 2 then the input was invalid
+    if (found_slashes == 2 && split_position + 1 <= input_char_length) {
+        char result_char[input_char_length - split_position];
```

Comment:
It looks like we need to make sure this value isn't 0? Windows warnings as error needs to be fixed but outside of that, looks good.